### PR TITLE
Core/Misc: Some improvements

### PR DESCRIPTION
**Pull request by r4d1sh**
Bug which was fixed by this PR:
1. Player X send summon request to player Y
2. Player Z send summon request to player Y
3. Player Y can't accept summon

Now if player Y already has active summon request from player X he will not receive new summon requests.

### CODE REVIEW

**Comment by tkrokli** [comment]
I presume that this PR replaces #17048 since the title is the same and that you closed the previous one 5 minutes before opening this one. The number of files you have changed are different, although the files you have included in this one are also part of the old PR. With no description in your first post, your intentions are somewhat uncertain. (I just wanted to mention this and link this PR back to the old one, which was closed without a descriptive closing comment.)

**Comment by Tonghost** [comment]
@tkrokli d5f04f5 already merged but present in #17048 so I decided open new PR

**Comment by tkrokli** [comment]
OK, good. That explains the changes you made. Thanks for posting the information. :+1:

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -23209,13 +23190,10 @@ void Player::UpdateForQuestWorldObjects()
     GetSession()->SendPacket(&packet);
 }
 
-void Player::SetSummonPoint(uint32 mapid, float x, float y, float z)
+void Player::SetSummonPoint(WorldLocation location)
```
Comment (et65) [inline src/server/game/Entities/Player/Player.cpp]: Better to use reference: WorldLocation const& location :)

**Comment by jackpoz** [comment]
> No description provided.

**Comment by Tonghost** [comment]
Update description

File: src/server/game/Entities/Player/Player.cpp [hunk 2]
```diff
@@ -23209,13 +23190,27 @@ void Player::UpdateForQuestWorldObjects()
...
+void Player::SendSummonRequestFrom(Unit* summoner)
 {
+    if (!summoner)
+        return;
+
+    // Player already has active summon request
+    if (m_summon_expire >= time(nullptr))
+        return;
+
+    // Evil Twin (ignore player summon, but hide this for summoner)
```
Comment (Shauren) [inline src/server/game/Entities/Player/Player.cpp]: what is this?

**Comment by P-Kito** [comment]
On retail 2 players click the portal and the player can accept as long as they are channling.

Maybe needs a check for 3.3.5 too, not sure how it was back then.

**Comment by Tonghost** [comment]
@P-Kito, I doubt that the time during which it is possible to accept summon request somehow depends on spell channeling at least for 3.3.5.

This PR ready to merge If there are no critical remarks.

File: src/server/game/Server/WorldSocketMgr.h [hunk 3]
```diff
@@ -35,7 +35,7 @@ class TC_GAME_API WorldSocketMgr : public SocketMgr<WorldSocket>
     typedef SocketMgr<WorldSocket> BaseSocketMgr;
 
 public:
-    static WorldSocketMgr& Instance();
+    static WorldSocketMgr* instance();
```
Comment (jackpoz) [inline src/server/game/Server/WorldSocketMgr.h]: what has this change to do with anything else in this PR ?

**Comment by Shauren** [comment]
Merged in 1aa30d3cce8a2003ad7873410fe9ec09ff357dcf

**Pull request closed without merging**